tests/native-lib/pass/scalar_arguments.rs

```diff
@@ -39,5 +39,54 @@ fn main() {
 
         // test void function that prints from C
         printer();
+
+        test_pass_struct();
+        test_pass_struct_complex();
+    }
+}
+
+/// Test passing a basic struct as an argument.
+fn test_pass_struct() {
+    #[repr(C)]
+    struct PassMe {
+        value: i32,
+        other_value: i16,
     }
+
+    extern "C" {
+        fn pass_struct(s: PassMe) -> i32;
+    }
+
+    let pass_me = PassMe { value: 42, other_value: 1337 };
+    assert_eq!(unsafe { pass_struct(pass_me) }, 42 + 1337);
+}
+
+/// Test passing a more complex struct as an argument.
+fn test_pass_struct_complex() {
+    #[repr(C)]
+    struct ComplexStruct {
+        part_1: Part1,
+        part_2: Part2,
+        part_3: u32,
+    }
+    #[repr(C)]
+    struct Part1 {
+        high: u16,
+        low: u16,
+    }
+    #[repr(C)]
+    struct Part2 {
+        bits: u32,
+    }
+
+    extern "C" {
+        fn pass_struct_complex(s: ComplexStruct) -> i32;
+    }
+
+    let complex = ComplexStruct {
+        part_1: Part1 { high: 0xabcd, low: 0xef01 },
+        part_2: Part2 { bits: 0xabcdef01 },
+        part_3: 0xabcdef01,
+    };
+    assert_eq!(unsafe { pass_struct_complex(complex) }, 0);
```

Comment:
Can you add a fail test that looks roughly like this, to ensure we do check initialization properly?
```rust
let arg = MaybeUninit::<ComplexStruct>::uninit();
pass_struct_complex(*arg.as_ptr());
```
This needs `derive(Copy, Clone)` on the struct. (Don't use `.read()`, that might defeat the test.)